src/Kiota.Builder/Plugins/PluginsGenerationService.cs

```diff
@@ -590,4 +591,33 @@ rExtRaw is T rExt &&
 
         return null;
     }
+
+    private static FunctionCapabilities? GetFunctionCapabilitiesFromOperation(OpenApiOperation openApiOperation)
+    {
+        var responseSemantics = GetResponseSemanticsFromAdaptiveCardExtension<OpenApiAiAdaptiveCardExtension>(openApiOperation, OpenApiAiAdaptiveCardExtension.Name);
+        if (responseSemantics is null)
+            return null;
+        return new FunctionCapabilities
+        {
+            ResponseSemantics = responseSemantics,
+        };
+    }
+
+    private static ResponseSemantics? GetResponseSemanticsFromAdaptiveCardExtension<T>(OpenApiOperation openApiOperation, string extensionName)
+    {
+        if (openApiOperation.Extensions is not null &&
+            openApiOperation.Extensions.TryGetValue(extensionName, out var adaptiveCardExtension) && adaptiveCardExtension is OpenApiAiAdaptiveCardExtension adaptiveCard)
+        {
+            string jsonString = $"{{\"file\":\"{adaptiveCard.File}\"}}";
+            using JsonDocument doc = JsonDocument.Parse(jsonString);
```

Comment:
```suggestion
            JsonNode node = new JsonObject();
            node["file"] = JsonValue.Create(adaptiveCard.File);
            using JsonDocument doc = JsonDocument.Parse(node.ToJsonString());
```

Just a suggestion, but you can use this rather than do string interpolation here. 
This way, in the unlikely scenario where `adaptiveCard.File` is null, you will have the json correctly written as 
`{"file": null}` rather than `{"file":"null"}`(null will be string interpolated and potentially misinterpreted as a value)